tests/signal/test_signal_service.py

```diff
@@ -275,6 +275,67 @@ def test_filter_actions_deduplicate(session, signal, project):
     assert signal_instance_2.filter_action == SignalFilterAction.deduplicate
 
 
+def test_filter_action_with_dedupe_and_snooze(session, signal, project):
+    from datetime import datetime, timedelta, timezone
+    from dispatch.signal.models import (
+        SignalFilter,
+        SignalInstance,
+        SignalFilterAction,
+    )
+    from dispatch.signal.service import filter_signal
+    from dispatch.entity_type.models import EntityType
+    from dispatch.entity.models import Entity
+
+    entity_type = EntityType(
+        name="dedupe1+snooze",
+        jpath="id",
+        regular_expression=None,
+        project=project,
+    )
+    session.add(entity_type)
+
+    entity = Entity(name="dedupe1+snooze", description="test", value="foo", entity_type=entity_type)
+    session.add(entity)
+
+    # create instance
+    signal_instance_1 = SignalInstance(
+        raw=json.dumps({"id": "foo"}), project=project, signal=signal, entities=[entity]
+    )
+    session.add(signal_instance_1)
+
+    signal_instance_2 = SignalInstance(
+        raw=json.dumps({"id": "foo"}), project=project, signal=signal, entities=[entity]
+    )
+    session.add(signal_instance_2)
+    session.commit()
+    # create deduplicate signal filter
+    signal_filter = SignalFilter(
+        name="dedupe1",
+        description="test",
+        expression=[
+            {"or": [{"model": "EntityType", "field": "id", "op": "==", "value": entity_type.id}]}
+        ],
+        action=SignalFilterAction.deduplicate,
+        window=5,
+        project=project,
+    )
+    signal.filters.append(signal_filter)
+
+    signal_filter = SignalFilter(
+        name="snooze0",
+        description="test",
+        expression=[{"or": [{"model": "Entity", "field": "id", "op": "==", "value": entity.id}]}],
+        action=SignalFilterAction.snooze,
+        expiration=datetime.now(tz=timezone.utc) + timedelta(minutes=5),
+        project=project,
+    )
+    signal.filters.append(signal_filter)
+
+    session.commit()
+    assert filter_signal(db_session=session, signal_instance=signal_instance_2)
+    assert signal_instance_2.filter_action == SignalFilterAction.snooze
```

Comment:
```suggestion
    # we assert the signal was snoozed, even though a deduplication filter was set, and set first
    assert signal_instance_2.filter_action == SignalFilterAction.snooze
```

theres some nuance thats helpful to explain here, i think